src/plot_api/helpers.js

```diff
@@ -102,13 +102,12 @@ exports.cleanLayout = function(layout) {
         }
     }
 
-    if(layout.annotations !== undefined && !Array.isArray(layout.annotations)) {
-        Lib.warn('Annotations must be an array.');
-        delete layout.annotations;
-    }
     var annotationsLen = (layout.annotations || []).length;
```

Comment:
in `handleArrayContainerDefaults` you do `Array.isArray(parentObjIn[name]) ? parentObjIn[name] : []` - you want to do that here too so we still keep going if `annotations` isn't even an array?